### ISSUE TITLE
gemspec file

### ISSUE THREAD

**Issue by MSP-Greg** [ruby/digest#1]
@hsbt  Thanks for the work (gems & core).

1. It might be helpful if digest had a constant string that was the version of OpenSSL it was compiled with.  Just thinking of when OpenSSL 1.1.0 becomes more common...

2. Reminder - should something like the following be added to gemspec
```
  spec.extensions    = %w[
    ext/digest/extconf.rb
    ext/digest/bubblebabble/extconf.rb
    ext/digest/md5/extconf.rb
    ext/digest/rmd160/extconf.rb
    ext/digest/sha1/extconf.rb
    ext/digest/sha2/extconf.rb
  ]
  spec.metadata["msys2_mingw_dependencies"] = "openssl"
```

I'm trying to build (not fat/binary) and install as a gem, then using the core digest tests.

When building with ruby-loco, my build `.ext/x64-mingw32/digest` directory has `bubblebabble.so`, `md5.so`, `rmd160.so`, `sha1.so`, and `sha2.so` are all around 323 KB, but the so files in the pkg/install `lib\ruby\2.5.0\x64-mingw32\digest` directory are all around 20 KB.

When trying to build/install the gem, the files stay at the large size, and also fail a few tests.

Obviously, I'm missing something.  I know a lot more about c build systems than I did six months ago, but I've got a long way too go.  FYI, ruby-loco is now working with the ruby repo set anywhere past 2.3.0, although I haven't decided the best way to handle spec/mspec.  I need to clean things up and push back up to GitHub.  [appveyor_ruby](https://ci.appveyor.com/project/MSP-Greg/appveyor-ruby) shows info on a 2.3.4 build and my most recent trunk build, along with several default Appveyor builds...

**Comment by hsbt** [comment]
Thanks for your reminder.

I fixed 0d4a102ddf3c2b9e0562735b945a93fa22f80b6c and f71baeb45ca05c82ced47953a2030f88a24cb321

**Comment by MSP-Greg** [comment]
Thanks.  I apologize, I just assumed a ruby repo meant Redmine/svn.  Next time I could probably do the PR myself...

Re my babbling, do *nix or OSX systems strip .so files when building binary gems?
